geoips/plugins/modules/output_checkers/image.py

```diff
@@ -142,14 +144,30 @@ def outputs_match(plugin, output_product, compare_product, threshold=0.05):
         message = f"compare_product: {np.array(comp_img).shape} {compare_product}"
         log_with_emphasis(LOG.interactive, message)
         return False
+    # Don't bother doing pixelmatch if the arrays are identical.
+    if diff_arr.min() == diff_arr.max():
```

Comment:
Question-- is this resilient against masked values? IE if all pixels are identical value but the comp_img has more masked points than the other this will say the comparisons are good when in reality they are not.

This might not be applicable to the library being used, though. Admit I don't know a lot about the nuts and bolts of how it works.

Reply:
That is interesting - I don't think the "Image" arrays are masked, I'm sure they are RGBA ?  They are numpy ndarrays.  But not sure about nan's...  I'll look at this a bit more.

Reply:
I updated this to diff_arr.min() == 0 and diff_arr.max() == 0, this should at least fail if there are any nans, which would cause it to continue to the pixelmatch tests.  So at least we won't get false passes.

Reply:
Package: geoips_full
Total run time: 2036 seconds
Number data types run: 56
Number data types failed: 0
Wed Jul 31 19:14:53 UTC 2024

Still get 0 return.